### PR TITLE
Split apart `FrameTimeDiagnosticsPlugin`

### LINKED ISSUE #7776: Fine tuning FrameTimeDiagnosticsPlugin
## What problem does this solve or what need does it fill?
Currently, adding the `FrameTimeDiagnosticsPlugin` will automatically log frame time, fps and frame count.
However, most of the time I am only interested in one of these options.
You could of course let the user write his own tiny system, but I think this is something that should be easily doable in bevy

## What solution would you like?
I'd still argue that the default should stay as is.
I would enable the user to use the plugin as follows
```rust
App::new()
  // adds everything as before
  .add_plugin(FrameTimeDiagnosticsPlugin::default())
  // only adds frame time
  .add_plugin(FrameTimeDiagnosticsPlugin::log_frame_time())
  // only adds fps
  .add_plugin(FrameTimeDiagnosticsPlugin::log_fps())
  // only adds frame count
  .add_plugin(FrameTimeDiagnosticsPlugin::log_frame_time())
````
## What alternative(s) have you considered?
Also I feel like the amount history size should also be easily configurable.
Which I would propose as a slight variation of the first solution using a builder.
```rust
App::new()
  // default
  .add_plugin(FrameTimeDiagnosticsPlugin::default())
  // with default history size
  .add_plugin(FrameTimeDiagnosticsPlugin::from_fps(FpsLogger::default()))
  // with configured history size
  .add_plugin(FrameTimeDiagnosticsPlugin::from_fps(FpsLogger::history_size(100)))
  // ... also for other 2 options
```
Even though, I think this would make it even more configurable, I honestly am not sure if it is to verbose.

## Additional context
**I would gladly take over the part of implementing this,**
but would be interested if this behavior is even welcomed/needed.

### PR DESCRIPTION
**Pull request by EmiOnGit**
# Objective
Allowing more fine graded control over the diagnostics plugin
Fixes #7776 

## Solution

- Describe the solution used to achieve the objective above.
The `FrameTimeDiagnosticsPlugin` is now a plugin group named `FrameTimeDiagnosticsPlugins`
containing the plugins:
```
FpsDiagnosticsPlugin
FrameTimeDiagnosticsPlugin
FrameCountDiagnosticsPlugin
```
---

## Migration Guide
The `FrameTimeDiagnosticsPlugin` is now a plugin group named `BasicPerformanceDiagnosticsPlugins`
```rust
// Old
App::new()
    .add_plugin(FrameTimeDiagnosticsPlugin)
// New
App::new()
    .add_plugins(BasicPerformanceDiagnosticsPlugins)
```

### CODE REVIEW

**Comment by EmiOnGit** [comment]
hum seems like it failed again, 
however I can not really see the mistake and it is hard for me to find the right link since my computer crashes running the ci tests.
Would be really glad if someone could lend a hand!

File: crates/bevy_diagnostic/src/frame_time_diagnostics_plugin.rs
Comment (alice-i-cecile) [inline crates/bevy_diagnostic/src/frame_time_diagnostics_plugin.rs:13]: I like this direction, but agree on the need for a clearer name. `BasicPerformanceDiagnosticsPlugins` perhaps?

File: crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs
Comment (alice-i-cecile) [inline crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:11]:
```suggestion
```

**Comment by EmiOnGit** [comment]
Sure! Will add some missing docs

File: crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs
Comment (EmiOnGit) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs]: Has been changed in [6d8ec89](https://github.com/bevyengine/bevy/pull/7780/commits/6d8ec89d885d90b4fd0e017ea246eae218edcbf8)

**Comment by EmiOnGit** [comment]
The docs have been added (also made sure `#[warn(missing_docs)]` doesn't complain)
Thank you @alice-i-cecile for your guidance! :smile: 
If you see something that still needs change just write another comment

File: crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs
Comment (alice-i-cecile) [inline crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:25]: We should avoid reusing this name; it'll lead to subtly broken "automatic" migration for users in the change over.

I would be okay with just naming these `FrameTimePlugin` and so on.

Comment (EmiOnGit) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:25]: Would the intend of an `FpsPlugin` be obvious to user?
How about `FpsMeasurePlugin` ?

Comment (alice-i-cecile) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:25]: Yeah I'm fine with that. We can bikeshed the name(s) as we see fit: the key thing is avoiding the clash.

Comment (EmiOnGit) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:25]: Do agree that avoiding to use the names is a good idea though.
I'm simply not sure what the right names would be.
However, if you're confident about `FrameTimePlugin`, `FpsPlugin` and `FrameCountPlugin` I'll change them

Comment (alice-i-cecile) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs:25]: I think that's the simplest choice, and it's very easy to read the docs, especially with an IDE tooltip on hover.

Comment (EmiOnGit) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs]: Seems like I didn't receive your messages in time. I'll revert my changes tomorrow :)

Comment (EmiOnGit) [reply crates/bevy_diagnostic/src/basic_performance_diagnostics_plugin.rs]: Now renamed again to your suggestion `FrameTimePlugin`, `FpsPlugin` and `FrameCountPlugin`

**Review by MinerSebas — CHANGES_REQUESTED** [review]
Please update the Migration Guide in the Pr Description, to reflect the renamed Plugins.

**Review by alice-i-cecile — APPROVED** [review]
Code LGTM now. Seconding the migration guide comment, and requesting a more informative PR title (they become commit names) :)

Perhaps "Split apart `FrameTimeDiagnosticsPlugin`"?

**Comment by EmiOnGit** [comment]
The migration guide only needs to contain the change of the `FrameTimeDiagnosticsPlugin` right? :)

**Comment by alice-i-cecile** [comment]
> The migration guide should only contain the change of the `FrameTimeDiagnosticsPlugin` right? :)

Yep, the listed migration guide is perfect: it helps users understand how to make their existing code work again.

**Comment by EmiOnGit** [comment]
Alright, then I suppose everything is ready for merging right?

**Comment by alice-i-cecile** [comment]
Yep, just need a second reviewer to approve :) I may also hold off until after 0.10 here, just to avoid making more work for the patch notes.

**Comment by mockersf** [comment]
From the linked issue:

> Currently, adding the FrameTimeDiagnosticsPlugin will automatically log frame time, fps and frame count.
> However, most of the time I am only interested in one of these options.

This is already possible, the `LogDiagnosticsPlugin` has a filter to only select some diagnostics to log: https://docs.rs/bevy/latest/bevy/diagnostic/struct.LogDiagnosticsPlugin.html#method.filtered

**Comment by EmiOnGit** [comment]
> This surely is handy too, but as you add more and more Diagnostics, you'd have to add them to the filter one by one right?
As response from the issue.

I actually don't mind if the merge request gets declined or other changes are needed. 
I think that it would be a good idea to decide soon before the merge request get's buried to never be looked at again 
@alice-i-cecile

**Comment by alice-i-cecile** [comment]
Yeah that's valid. We'll close this out for now then :)

**Pull request closed without merging**
